### PR TITLE
interp_nan-added

### PR DESCRIPTION
**Pull request by gfudenberg**

### CODE REVIEW

File: cooltools/lib/numutils.py [hunk 1]
```diff
@@ -141,6 +141,64 @@ def fill_nainf(arr, value=0, copy=True):
...
+        if verbose==True: print('interpolating 1D vector')
+        a_init = a_init.ravel()
+        if pad_zeros == True:
+            A = np.zeros(len(a_init)+2,)
+            A[1:-1] = a_init
+        else:
+            A = a_init
+        x = np.arange(len(A))
+        interp = np.array(A)
+        interp[np.isnan(A)] = scipy.interpolate.interp1d(
```
Comment (nvictus) [inline cooltools/lib/numutils.py]: You may need to import `scipy.interpolate` at the top of the file.

File: cooltools/lib/numutils.py [hunk 2]
```diff
@@ -141,6 +141,64 @@ def fill_nainf(arr, value=0, copy=True):
...
+            a[1:-1,1:-1] = a_init
+        else:
+            a = a_init
+        x, y = np.indices(a.shape)
+        interp = np.array(a)
+        interp[np.isnan(interp)] = scipy.interpolate.griddata(
+                 (x[~np.isnan(a)], y[~np.isnan(a)]), # points we know
+                  a[~np.isnan(a)],                    # values we know
+                 (x[np.isnan(a)], y[np.isnan(a)]) ,method='linear')
+        if pad_zeros == True:
```
Comment (nvictus) [inline cooltools/lib/numutils.py]: Style: no need for == True when checking a boolean. `if pad_zeros:`

File: cooltools/lib/numutils.py [hunk 3]
```diff
@@ -141,6 +141,64 @@ def fill_nainf(arr, value=0, copy=True):
     arr[~np.isfinite(arr)] = value
     return arr
 
+def interp_nan(a_init, pad_zeros=True, verbose=False):
```
Comment (nvictus) [inline cooltools/lib/numutils.py]: Style: double space between function definitions.

File: cooltools/lib/numutils.py [hunk 4]
```diff
@@ -141,6 +141,64 @@ def fill_nainf(arr, value=0, copy=True):
...
+        if verbose==True: print('interpolating 2D matrix')
+        if pad_zeros == True:
+            a = np.zeros((init_shape[0]+2,init_shape[1]+2))
+            a[1:-1,1:-1] = a_init
+        else:
+            a = a_init
+        x, y = np.indices(a.shape)
+        interp = np.array(a)
+        interp[np.isnan(interp)] = scipy.interpolate.griddata(
+                 (x[~np.isnan(a)], y[~np.isnan(a)]), # points we know
```
Comment (nvictus) [inline cooltools/lib/numutils.py]: Pre-computing the `~np.isnan(a)` mask could speed things up a bit.

Comment (nvictus) [reply cooltools/lib/numutils.py]: It should even be `np.isnan(interp)`

Comment (gfudenberg) [reply cooltools/lib/numutils.py]: replaced with a much faster method based on scipy.interpolate.RegularGridInterpolator

File: cooltools/lib/numutils.py (hunk 1 above)
Comment (gfudenberg) [reply cooltools/lib/numutils.py]: right!

**Pull request merged**